src/WidgetBase.ts

```diff
@@ -241,6 +247,20 @@ export class WidgetBase<P extends WidgetProperties = WidgetProperties, C extends
 		this._checkOnElementUsage();
 	}
 
+	protected meta<T extends MetaBase>(MetaType: WidgetMetaConstructor<T>): T {
+		let cached = this._metaMap.get(MetaType);
+		if (!cached) {
+			cached = new MetaType({
+				nodes: this._nodeMap,
+				requiredNodes: this._requiredNodes,
+				invalidate: this.invalidate.bind(this)
+			});
+			this._metaMap.set(MetaType, cached);
+		}
+
+		return cached as T;
```

Comment:
Because `T` extends `MetaBase`, when I just say `let cached: T = ` it complains that `MetaBase is not assignable to T` because `T` is a subclass of `MetaBase`.